src/code_80091750.c

```diff
@@ -1515,9 +1516,9 @@ void func_80091B78(void) {
     gNextFreeMemoryAddress = gFreeMemoryResetAnchor;
     // Hypothetically, this should be a ptr... But only hypothetically.
     D_8018D9B0 = get_next_available_memory_addr(0x000900B0);
-    D_8018D9B4 = (uintptr_t *) get_next_available_memory_addr(0x0000CE00);
-    D_8018D9B8 = (uintptr_t *) get_next_available_memory_addr(0x00012C00);
-    D_8018D9C0 = (struct_8018EE10_entry *) get_next_available_memory_addr(0x00001000);
+    D_8018D9B4 = (u8 *) get_next_available_memory_addr(0x0000CE00);
+    D_8018D9B8 = (u8 *) get_next_available_memory_addr(0x00012C00);
```

Comment:
This is the only other concern I had. I think your change is probably fine. Just need a sec to look into this to confirm.

Reply:
Took a look, I like the u8 * change